### PR TITLE
Fix Fatal error on invalid image

### PR DESCRIPTION
**Pull request by marijnkampf**
If user has uploaded an invalid image the original code would crash when called from template

### CODE REVIEW

File: forms/UploadField.php [hunk 1]
```diff
@@ -870,11 +870,13 @@ protected function getThumbnailURLForFile(File $file) {
 			$width = $this->getPreviewMaxWidth();
 			$height = $this->getPreviewMaxHeight();
 			if ($file->hasMethod('getThumbnail')) {
-				return $file->getThumbnail($width, $height)->getURL();
+				$r = $file->getThumbnail($width, $height)
```
Comment (dhensby) [inline forms/UploadField.php]: Missing semicolon

**Comment by jonom** [comment]
Looks good to me but can you squash your commits please?

**Comment by marijnkampf** [comment]
Not quite sure how to squash them as I made the changes directly on github.com. Checked out a local copy of my branch, but can't figure out how to add it to the pull request as it says it's up to date when trying to commit...

**Comment by SpiritLevel** [comment]
Hi @marijnkampf 

Have a look at https://docs.silverstripe.org/en/3.2/contributing/code/ for git usage...

Summary: Clone your github fork of the silverstripe repo to you local machine. Locally, checkout patch-6 and use the command 'git rebase -i upstream/master' to squash; you'll get an editor. In all but the top commit line, replace 'pick' with 'squash' and then save. Leave the top line as 'pick'. Use 'git log' to see if all your individual commits got combined into one. Then, use 'git push -f origin patch-6' to send your local patch-6 branch to your github account. This will update any pull request made using your github account to have patch-6 merged into the silverstripe repo. If you haven't already made the pull request for patch-6, go to the silverstripe repo to do so :) 

Note: upstream refers to the silverstripe repo and origin refers to the fork of the repo in your github account, which you clone. Use 'git config --list' to verify the upstream and origin are set properly. When you clone your github fork, it should automatically configure git so that origin is your github repo. Hope that helps :) D.

**Comment by dhensby** [comment]
you should be able to do:

```
git checkout patch-6 # checkout the PR branch
git reset --hard 229a0a7 # reset back to your last commit (you've added a merge that we don't want)
git reset HEAD^ # reset to the parent commit
git commit -a --amend --no-edit # modify the last commit with the current changes (add the ;)
git push -f origin patch-6 # forch push up to your PR branch
```

**Comment by marijnkampf** [comment]
Thank you both for the instructions. I've followed Dan's step-by-step, which hopefully has done the trick!

**Comment by dhensby** [comment]
thanks!

**Comment by SpiritLevel** [comment]
Sorry @marijnkampf ! I clearly missed the boat with my "answer"...

@dhensby In your step-by-step, where did you find the 229a0a7 reference ? I couldn't see it anywhere. 

An explanation of how to locally manipulate one's changes already made directly on github would be a useful addition to the docs, don't you think ?

**Comment by dhensby** [comment]
@SpiritLevel the hash was referenced in the list of the commits on the PR.



That would be a useful addition, but would probably be quite a big addition as it's expected that users who make edits via GitHub website likely don't have the CLI tools installed locally...

**Comment by SpiritLevel** [comment]
Sorry mate, slight git confusion here...I see 006717a in your screen shot but 229a0a7 in your instructions...should that be 006717a in the reset --hard command ? Or is the 229a0a7 hash no longer visible after squashing ? I'm thinking its the latter.

**Comment by dhensby** [comment]
Yes, exactly, it's no longer visible because the squashing re-wrote the git history and so the commit hash has changed.

My screenshot was just to show where I saw the hash (which has now changed/been removed since the sqaushing).

**Comment by SpiritLevel** [comment]
Perfect :) Thanks @dhensby...helpful, as usual. As for instructions, perhaps one could put a comment in the appropriate docs section explaining that when making edits via Github website one should only make a single commit. If multiple commits are made using Github website the reader can then be directed to the section on code contribution where CLI git use is detailed, squashing in particular, and just add a couple of paragraphs to that explaining the steps you gave above. This would keep the addition small. Maybe I'll take a stab at this....

**Pull request merged**